### PR TITLE
Enabled APK mailing on app generator 

### LINKED ISSUE #1601: Droidgen.eventyay doesn't mail the APK
**Actual Behaviour**
No mail is sent with the generated APK

**Expected Behaviour**
Earlier, it used to send the mail.

### PR DESCRIPTION
**Pull request by mahikaw**
Fixes #1601 

Changes: Enabled app generator(v2) to mail APK via:

- Sendgrid, or
- SMTP 

-----

@niranjan94 @the-dagger @iamareebjamal @Shailesh351 @shreyasnbhat @rohanagarwal94 @arp95 Please review. 

Also, @niranjan94 Could you please specify the environment variables, if any that I should parse using ```envparse``` so that I can make the required changes.

### CODE REVIEW

File: apk-generator/v2/app/utils/notification.py
Comment (niranjan94) [inline apk-generator/v2/app/utils/notification.py:25]: This function is just for choosing which method to use. SMTP related config to be done in the smtp send method. Sendgrid config to be done sendgrid method.

File: apk-generator/v2/app/utils/notification.py
Comment (niranjan94) [inline apk-generator/v2/app/utils/notification.py]: No `delay` here

File: apk-generator/v2/app/utils/notification.py
Comment (niranjan94) [inline apk-generator/v2/app/utils/notification.py]: Why are you assigning `message` to a property of `message` ? Should be the message from the payload.

File: apk-generator/v2/config.py
Comment (niranjan94) [inline apk-generator/v2/config.py]: Space around `=` for all of this.

Comment (niranjan94) [inline apk-generator/v2/config.py]: Call env variable also as `EMAIL_SERVICE` and keep default as sendgrid.

File: apk-generator/v2/config.py
Comment (niranjan94) [inline apk-generator/v2/config.py]: Call env variable also as `SMTP_ENCRYPTION`

File: apk-generator/v2/config.py
Comment (niranjan94) [inline apk-generator/v2/config.py]: Call both variable and env variable as `SMTP_HOST`

File: apk-generator/v2/config.py
Comment (niranjan94) [inline apk-generator/v2/config.py]: Call both variable and env variable as `SMTP_USERNAME`

File: apk-generator/v2/config.py
Comment (niranjan94) [inline apk-generator/v2/config.py]: Call both variable and env variable as `SMTP_PASSWORD`

File: apk-generator/v2/config.py
Comment (niranjan94) [inline apk-generator/v2/config.py]: Call both variable and env variable as `SMTP_PORT`

File: apk-generator/v2/config.py
Comment (niranjan94) [inline apk-generator/v2/config.py]: What is this ?

File: apk-generator/v2/config.py
Comment (niranjan94) [inline apk-generator/v2/config.py]: Call both variable and env variable as `SENDGRID_KEY `

File: apk-generator/v2/app/utils/notification.py
Comment (niranjan94) [inline apk-generator/v2/app/utils/notification.py]: Why the brackets ?

**Review by niranjan94 — CHANGES_REQUESTED** [review]
@mahikaw

**Comment by mahikaw** [comment]
Updated @niranjan94 Please review.

File: apk-generator/v2/config.py
Comment (iamareebjamal) [inline apk-generator/v2/config.py]: Add space after comma

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

File: apk-generator/v2/app/utils/notification.py
Comment (niranjan94) [inline apk-generator/v2/app/utils/notification.py]: The smtp username may not always be an email address. And why would there be an smtp username if the the user wants to use sendgrid. ? 

So, have a separate variable called as `FROM_EMAIL`

File: apk-generator/v2/app/utils/notification.py
Comment (niranjan94) [inline apk-generator/v2/app/utils/notification.py]: Use the logger to log stuff

File: apk-generator/v2/app/utils/notification.py
Comment (niranjan94) [inline apk-generator/v2/app/utils/notification.py]: call this `send_email_via_sendgrid`

File: apk-generator/v2/app/utils/notification.py
Comment (niranjan94) [inline apk-generator/v2/app/utils/notification.py]: The `message` is a rich text string. (Has html tags). So, have both `message.plain` (with html tags stripped) and `message.rich`

File: apk-generator/v2/app/utils/notification.py
Comment (niranjan94) [inline apk-generator/v2/app/utils/notification.py]: Update this comment

**Review by niranjan94 — CHANGES_REQUESTED** [review]
@mahikaw

**Comment by mahikaw** [comment]
@niranjan94 @iamareebjamal , updated it.

File: apk-generator/v2/app/utils/notification.py
Comment (niranjan94) [inline apk-generator/v2/app/utils/notification.py]: I meant ... API implementation is still remaining, so keep the TODO for that

File: apk-generator/v2/app/utils/notification.py
Comment (niranjan94) [inline apk-generator/v2/app/utils/notification.py]: Send via smtp or sendgrid, only if `via_api` is `False`

File: apk-generator/v2/app/utils/notification.py
Comment (niranjan94) [inline apk-generator/v2/app/utils/notification.py]: There is no param called `html`. Its called as `message`. Put that in rich and use strip tags function on it and use that for plain.

**Review by niranjan94 — CHANGES_REQUESTED** [review]

File: apk-generator/v2/config.py
Comment (niranjan94) [inline apk-generator/v2/config.py]: Space after `=` please

**Review by niranjan94 — CHANGES_REQUESTED** [review]

File: apk-generator/v2/app/utils/notification.py
Comment (mahikaw) [reply apk-generator/v2/app/utils/notification.py]: um yes. Sorry for not taking care.

File: apk-generator/v2/config.py
Comment (mahikaw) [reply apk-generator/v2/config.py]: Updating 👍

**Comment by mahikaw** [comment]
Updated @niranjan94

File: apk-generator/v2/app/utils/notification.py
Comment (niranjan94) [reply apk-generator/v2/app/utils/notification.py]: Isn't resolved yet @mahikaw

**Comment by mahikaw** [comment]
@iamareebjamal @niranjan94 Please approve if there aren't any further changes required.

File: apk-generator/v2/config.py
Comment (niranjan94) [inline apk-generator/v2/config.py]: Keep default as `none`. If none, email service should be disabled.

File: apk-generator/v2/app/utils/notification.py
Comment (niranjan94) [inline apk-generator/v2/app/utils/notification.py]: Are all the additional `None` arguments required?

File: apk-generator/v2/app/utils/notification.py
Comment (niranjan94) [inline apk-generator/v2/app/utils/notification.py]: Why the underscore at the end?

File: apk-generator/v2/app/utils/notification.py
Comment (niranjan94) [inline apk-generator/v2/app/utils/notification.py]: Why the underscore at the end.?

**Review by niranjan94 — CHANGES_REQUESTED** [review]

File: apk-generator/v2/config.py
Comment (mahikaw) [reply apk-generator/v2/config.py]: In one of the earlier reviews, you suggested to set it to 'send grid'. I'll make it none then and add a condition to check that as well?

File: apk-generator/v2/app/utils/notification.py
Comment (mahikaw) [reply apk-generator/v2/app/utils/notification.py]: I had done this after seeing the Marrow mailer [documentation](https://github.com/marrow/mailer). I think they are required. The maintype can be remove though, I'll do that.

**Comment by niranjan94** [comment]
Yes... I had suggested that. But.. I think the generator itself should work
without the need for an email server. (no emails. Wait and download).

So, have a none option. And keep it default. Also update the Readme file
with details about email.

**Comment by mahikaw** [comment]
@niranjan94 Updated as per suggestions. Please review the changes.

File: apk-generator/v2/app/utils/notification.py
Comment (niranjan94) [reply apk-generator/v2/app/utils/notification.py]: @mahikaw see here https://github.com/marrow/mailer/blob/develop/marrow/mailer/message.py#L254 ... The other arguments are not needed. The defaults are `None` anyway. So remove them please. Once this is done, I think the PR can be merged. 

Good work :)

**Review by niranjan94 — CHANGES_REQUESTED** [review]

**Comment by mahikaw** [comment]
Updated @niranjan94 👍

**Review by niranjan94 — APPROVED** [review]

**Review by rohanagarwal94 — APPROVED** [review]

**Pull request merged**
